lib/protocol/encryption.go

```diff
@@ -488,9 +491,30 @@ func knownBytes(folderID string) []byte {
 	return []byte("syncthing" + folderID)
 }
 
+type FolderKeyGenerator struct {
+	mut   sync.Mutex
+	cache *lru.TwoQueueCache[folderKeyGeneratorCacheKey, *[keySize]byte]
+}
+
+var DefaultFolderKeyGenerator FolderKeyGenerator
```

Comment:
I built a bit of a global-aversion during some past changes - don't even remember what they were, but the aversion stayed. I'd like to store this in `connection.service` and pass it to the protocol connection constructors. I am happy to propose such a change, as I am asking for (maybe even not that little) busywork/refactoring with not very strong arguments :)  
Let me know if you'd appreciate that.

Reply:
There you go... It's a fairly infectious change, and as I'm now touching one hundred files you get some spurious gofmt changes as well.

Reply:
Yeah, that infectiousness was what I was suspecting... Thanks a lot!

And wow, it's coming all the way from the top o.O  
I think I'd have taken the lazier route and created it inside `connection.service` :)

Reply:
Can't do that, because the model also needs it for constructing tokens. That could have been two separate instances because they only hold a cache and some duplication would be OK, but if we ever make the key generation instance smarter for whatever reason that would be a footgun...